packages/datasource-mongoose/src/collection.ts

```diff
@@ -1,41 +1,214 @@
+    filter: PaginatedFilter,
+    projection: Projection,
+  ): Promise<RecordData[]> {
+    const lookupProjection = projection.union(
+      filter.conditionTree?.projection,
+      filter.sort?.projection,
+    );
+
+    const records = await this.model.aggregate([
+      ...this.buildBasePipeline(filter, lookupProjection),
+      ...ProjectionGenerator.project(projection),
+    ]);
+
+    return replaceMongoTypes(records);
   }
 
-  async list(): Promise<RecordData[]> {
-    throw new Error('not implemented');
+  async update(caller: Caller, filter: Filter, patch: RecordData): Promise<void> {
+    const records = await this.list(caller, filter, new Projection('_id'));
+    const ids = records.map(record => record._id);
+
+    if (!this.prefix) {
+      return this.model.updateMany({ _id: ids }, patch, { rawResult: true }).then(() => {});
+    }
+
+    // Clean patch
+    const schema = MongooseSchema.fromModel(this.model).getSubSchema(this.prefix);
+    let cleanPatch = { ...patch };
+    delete cleanPatch._id; // Virtual field
+    delete cleanPatch._pid; // Ignore _pids: they should not be editable from the frontend.
+    if (Object.keys(cleanPatch).length === 0) return;
+    if (schema.isLeaf) cleanPatch = cleanPatch.content;
+
+    // Perform update
+    const idsByPath = groupIdsByPath(ids);
+    const promises = Object.entries(idsByPath).map(async ([path, rootIds]) =>
+      this.model.updateMany({ _id: rootIds }, { [path]: cleanPatch }, { rawResult: true }),
+    );
+
+    await Promise.all(promises);
   }
 
-  async update(): Promise<void> {
-    throw new Error('not implemented');
+  async delete(caller: Caller, filter: Filter): Promise<void> {
+    const records = await this.list(caller, filter, new Projection('_id'));
+    const ids = records.map(record => record._id);
+
+    if (!this.prefix) {
+      return this.model.deleteMany({ _id: ids }, { rawResult: true }).then(() => {});
+    }
+
+    const schema = MongooseSchema.fromModel(this.model).getSubSchema(this.prefix);
+    const idsByPath = groupIdsByPath(ids);
+
+    if (schema.isArray) {
+      for (const path of Object.keys(idsByPath).sort(compareIds).reverse()) {
+        const arrayPath = path.substring(0, path.lastIndexOf('.'));
+        const index = Number(path.substring(path.lastIndexOf('.') + 1));
+
+        // There is no update operator to pop items out of arrays at known positions
+        // => we use an aggregation pipeline in the update operation
+        // @see https://jira.mongodb.org/browse/SERVER-1014?focusedCommentId=2305681#comment-2305681
+        const newArrayValue = [
+          { $slice: [`$${arrayPath}`, index] },
+          { $slice: [`$${arrayPath}`, index + 1, { $size: `$${arrayPath}` }] },
+        ];
+
+        // When updating arrays, indexes will change with each request so we need to perform the
+        // request sequentially.
+        // eslint-disable-next-line no-await-in-loop
+        await this.model.collection.updateMany(
```

Comment:
```suggestion
        return this.model.collection.updateMany(
```
We win a else condition

Reply:
the call is in a for loop